src/Service/RequestForwarder.cpp

```
@@ -111,61 +135,55 @@ void RequestForwarder::runReceive(RunnerId runner_id)
-                        {
-                            LOG_ERROR(
-                                log,
-                                "Receive failed forward response with type(Result), session {}, xid {}, error code {}",
-                                response.session_id,
-                                response.xid,
-                                response.error_code);
-                            request_processor->onError(
-                                response.accepted,
-                                static_cast<nuraft::cmd_result_code>(response.error_code),
-                                response.session_id,
-                                response.xid,
-                                response.opnum);
-                        }
-                        else if (response.pkg_type == Session)
-                        {
-                            LOG_ERROR(
-                                log,
-                                "Receive failed forward response with type(Session), session {}, xid {}, error code {}",
-                                response.session_id,
-                                response.xid,
-                                response.error_code);
-                        }
-                        else if (response.pkg_type == Handshake)
-                        {
-                            LOG_ERROR(
-                                log,
-                                "Receive failed forward response with type(Handshake), session {}, xid {}, error code {}",
-                                response.session_id,
-                                response.xid,
-                                response.error_code);
-                        }
+                        continue;
                     }
+
+                    ForwardResponsePtr response;
+                    if (connection->receive(response))
+                        processResponse(runner_id, response);
```

When `connection->receive` return false? 

Maybe it is better `receive` throw exception instead of return `bool` when something else happens in which fashion the method result is more consistent.